### PR TITLE
Service Endpoint Auth Token generation

### PR DESCRIPTION
**Pull request by kmathur2**
- Added a flag in ServiceEndpoint object to determine if the endpoint needs auth token or not.
- If the above flag is set then generate auth token and store in member status as well as annotation for the k8s service object. 
- Added auth token in configmeta to be consumed by apps.

### CODE REVIEW

File: pkg/apis/kubedirector/v1beta1/kubedirectorapp_types.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ type ServiceEndpoint struct {
 	Port        *int32 `json:"port"`
 	Path        string `json:"path,omitempty"`
 	IsDashboard bool   `json:"isDashboard,omitempty"`
+	HasToken    bool   `json:"hasToken,omitempty"`
```
Comment (joel-bluedata) [inline pkg/apis/kubedirector/v1beta1/kubedirectorapp_types.go:98]: This should be authToken, not hasToken?

File: pkg/apis/kubedirector/v1beta1/kubedirectorcluster_types.go [hunk 2]
```diff
@@ -155,6 +155,7 @@ type RoleStatus struct {
 type MemberStatus struct {
 	Pod         string            `json:"pod"`
 	Service     string            `json:"service"`
+	AuthToken   string            `json:"authToken,omitempty"`
```
Comment (joel-bluedata) [inline pkg/apis/kubedirector/v1beta1/kubedirectorcluster_types.go:158]: The kdcluster CRD needs to be updated accordingly.

File: pkg/apis/kubedirector/v1beta1/kubedirectorapp_types.go (hunk 1 above)
Comment (joel-bluedata) [reply pkg/apis/kubedirector/v1beta1/kubedirectorapp_types.go:98]: Either that or the app CRD should be changed to say hasToken.

**Comment by joel-bluedata** [comment]
image build at bluek8s/kubedirector:da322dc FYI

**Comment by kmathur2** [comment]
> image build at bluek8s/kubedirector:da322dc FYI

👍 👍

**Pull request merged**